pkg/controller/controller.go

```diff
@@ -56,17 +60,21 @@ func New(
 	scheduledWorkQueue := scheduler.NewScheduledWorkQueue(clock.RealClock{}, workqueue.Add)
 
 	log = log.WithField("module", "controller")
+	nodesArchitecture := architecture.New()
 	versionGetter := version.New(log, imageClient, cacheTimeout)
 	search := search.New(log, cacheTimeout, versionGetter)
+	nodeController := nodes.New(log, nodesArchitecture)
 
 	c := &Controller{
 		log:                log,
 		kubeClient:         kubeClient,
 		workqueue:          workqueue,
 		scheduledWorkQueue: scheduledWorkQueue,
 		metrics:            metrics,
-		checker:            checker.New(search),
+		checker:            checker.New(search, nodesArchitecture),
 		defaultTestAll:     defaultTestAll,
+		nodesArchitecture:  nodesArchitecture,
+		nodeController:     nodeController,
```

Comment:
I'd like to see this controller split out into another pod controller, so you have;

` pkg/controller/controller.go` which instantiates the controllers
` pkg/controller/nodes/nodes.go` which implemented the node controller
` pkg/controller/pods/...` which implements the pods controller. Basically the main loops.

Happy to leave that in a follow up PR.

Reply:
yeah, if we could do it in a separate PR :sweat_smile: